### ISSUE TITLE
All packs unloaded with DashLoader

### ISSUE THREAD

**Issue by CuboidRaptor** [QuantumFusionMC/DashLoader-Definitions#20]
**Describe the bug**
When I load minecraft with DashLoader, all resource packs are unloaded. I believe this might be to do with CIT.

**To Reproduce**
1. Launch minecraft with DashLoader and a CIT pack (Hypixel+, in my case)
2. All resource packs are un-loaded.

**Expected behavior**
My resource packs load properly when minecraft launches,

**Context (please complete the following information):**
 - DashLoader Version [4.0.0 for 1.19]
 - Minecraft Version [e.g. 1.19]

Modlist, in case there are incompatibilities (I doubt it)
Advanced Compass
CIT Resewn (obviously)
Dynamic FPS
Fabric API (also obviously)
Inventory HUD+
Lazy DFU
Lithium
Mod Menu
Sodium Extras
Sodium
Starlight
ToroHealth Indicators
WI Zoom

Also, I can sometimes manually load my resource packs, however sometimes it breaks (seemingly randomly, although I'm sure there must be a pattern somewhere). However, I have a load of resource pack and that's kind of inconvienient.

**Comment by Agent00Ming** [comment]
I believe that this is a duplicate of issue #17

**Comment by Agent00Ming** [comment]
I am no longer experiencing this issue when using 'CIT Resewn' 1.1.1 on Minecraft 1.19. I suggest that you try it and update this issue accordingly as I have closed mine. (issue #17)

**Comment by CuboidRaptor** [comment]
I have tried, after creating the cache and relaunching it replaced all characters with boxes, before the text reverted to normal and all resource packs deloaded.
TL;DR same thing.
Are there any other changes you made to your minecraft client that might've fixed it instead? And, did you launch twice to ensure it would not break when loading from cache?

**Comment by Agent00Ming** [comment]
This is interesting. My optimized modded setup has no issues loading from cache and has yet to fail a single time after 20 or so times. However, when I created a light setup with the bare minimum to recreate the issue (CIT Resewn, Fabric API and Dashloader + CIT Resource pack) it has failed basically every single time I launch it.

**Comment by Agent00Ming** [comment]
OK, after about an hour of testing mod configurations, I've singled out the one factor that allows my modded setup to work properly: 
`[10:09:10] [main/ERROR]: Failed to load class. Mod: "enhancedblockentities", Value: "foundationgames.enhancedblockentities.compat.dashloader.DashDynamicBakedModel"`
A class that fails to load somehow changes something to fix/make the
`[10:13:02] [Render thread/INFO]: Caught error loading resourcepacks, removing all selected resourcepacks
java.util.concurrent.CompletionException: java.lang.IllegalStateException: Circular reference while loading minecraft:item/generated`
error go undetected.

**Comment by alphaqu** [comment]
Hi! This issue is caused because ebe is currently using 3.0 api and 4.0 changed our module paths which makes ebe crash.

**Comment by Agent00Ming** [comment]
CIT resource packs always cause
`[10:13:02] [Render thread/INFO]: Caught error loading resourcepacks, removing all selected resourcepacks java.util.concurrent.CompletionException: java.lang.IllegalStateException: Circular reference while loading minecraft:item/generated`
when launching from cache without ebe

**Comment by alphaqu** [comment]
Hi! Can you please provide full logs for when DashLoader tried loading the cache.

**Comment by alphaqu** [comment]
oh god please use pastebin

**Comment by Agent00Ming** [comment]
https://pastebin.com/NtNvwmvk

**Comment by CuboidRaptor** [comment]
I posted my logs in Discord, but I guess I'll repost it [here](https://pastebin.com/6TKvvnMg).
Should I also add a resource pack list?
